syntax = "proto2";
message AProduct{
    required int64 id = 1;
    required string description = 2;
    required int32 count = 3;
  }
  
  message AInitWarehouse{
    required int32 id = 1;
    required int32 x = 2;
    required int32 y = 3;
  }

  message AReqTruck {
    required AInitWarehouse warehouse = 1;    // warehouse info
    repeated AProduct product = 2;    // products info
    required int64 packageid = 3;    // packageID
    required int32 buyer_x = 4;
    required int32 buyer_y = 5;
    optional string ups_name = 6;
    required int64 sequenceNum = 7;
}

  
  message ACompleteLoading {
      required int32 truckid = 1;
      required int64 packageid = 2;
      required int64 sequenceNum = 3;
  }
  
  message AMsgs {
      repeated AReqTruck reqtruck = 1;
      repeated ACompleteLoading completeloading = 2;
      repeated int64 acks = 3;
      repeated Err err = 4;
  }
  
  message Err{
    required string err = 1;
    required int64 originseqnum = 2;
    required int64 seqnum = 3;
  }
  
  message UTruckArrived{
    required int32 truckid = 1;
    required int64 packageid = 2;
    required int64 seqnum = 3;
  }
  
  message UFinishDelivery{
    required int64 packageid = 1;
    required int64 seqnum = 2;
  }
  
  message UMsgs {
      repeated UTruckArrived trucks = 1;
      repeated UFinishDelivery finish = 2;
      repeated int64 acks =3; // responseTruck and responseLoaded
      repeated Err err = 4;
  }
  
